system-int-test/test/cmr/system_int_test/search/granule_spatial_search_test.clj

```diff
@@ -49,6 +51,48 @@
   [& ords]
   (codec/url-encode (umm-s/set-coordinate-system :geodetic (apply polygon ords))))
 
+;; Tests all granule search when allow-all-granule-params-flag is false
```

Comment:
Since we are not testing with the flag set to true, I think we need to add a bunch of tests to make sure we do not reject queries that search by each of the allowed granule restriction parameters (provider, collection concept ID, ...).

I'd also want a test that verifies setting most of the other parameters (like spatial, temporal, page_size, page_num, etc) still get rejected.

We can do most of the testing via unit tests of the function and just add a couple of integration tests for both rejection and non-rejection cases with the parameter set to false.

Reply:
Chris, great catch on this!  I was thinking about this and thought about setting the flag to true in the test file to safeguard it but then forgot all about it.  I think I might have to do that...

Reply:
Sorry, please disregard what I said. I didn't know what I was talking about. :)

Reply:
Done. Added all in integration tests.